File: sinapse_api/src/models/flashcard.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Flashcard {
    pub deck_id: String,
    pub question: String,
    pub answer: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ResponseFlashcard {
    pub id: String,
    pub deck_id: String,
    pub question: String,
    pub answer: String,
}

File: sinapse_bot/cogs/classes/flashcard.py
class Flashcard:
    def __init__(self, deck_id: str, question: str, answer: str) -> None:
        self.deck_id = deck_id
        self.question = question
        self.answer = answer

    def check_answer(self, answer: str) -> bool:
        return self.answer == answer